internal/controller/helmchart_controller_test.go

```diff
@@ -945,6 +945,50 @@ func TestHelmChartReconciler_buildFromHelmRepository(t *testing.T) {
 				g.Expect(os.Remove(build.Path)).To(Succeed())
 			},
 		},
+		{
+			name: "Missing values files are an error",
```

Comment:
In `name: "Uses artifact as build cache"`, please add `g.Expect(build.ValuesFiles).To(BeEmpty())` to `assertFunc`.
In `name: "Sets Generation as VersionMetadata with values files"`, add `g.Expect(build.ValuesFiles).To(Equal([]string{"values.yaml", "override.yaml"}))` to `assertFunc`.

Also a test case `name: "Uses artifact as build cache with observedValuesFiles"` would be nice.

`TestHelmChartReconciler_buildFromTarballArtifact` should have the same test cases.

Reply:
OK. I believe that's all done.